library/cpp/protocol/TBinaryProtocol.tcc

```diff
@@ -665,20 +657,12 @@ int32_t TBinaryProtocolT<Transport_>::readIPADDR(
   }
   if (version == AF_INET) {
     boost::asio::ip::address_v4::bytes_type ipv4;
-#ifdef _WINDOWS
     this->trans_->readAll(ipv4.data(), 4);
```

Comment:
`readAll` needs pointer to non-const data. `boost:array::data()` returns pointer to const data. Please check if it works with `boost::array`.